### PR TITLE
feat-6837: add helmfile documentation

### PR DESCRIPTION
**Pull request by thameezb**
Adds in documentation for helmfile addition to Spinnaker

Relies on https://github.com/spinnaker/deck/pull/9998

Related to https://github.com/spinnaker/spinnaker/issues/6837

### CODE REVIEW

**Comment by dbyron-sf** [comment]
For completeness, can you rebase to the HEAD of master please?

**Comment by dbyron-sf** [comment]
And also add something to the [next release preview](https://github.com/spinnaker/spinnaker.io/blob/1c4a755e6e69689d6dd493c042c1b777602686c7/content/en/docs/releases/next-release-preview/index.md) please?

File: content/en/docs/releases/next-release-preview/index.md [hunk 1]
```diff
@@ -71,3 +71,14 @@ https://github.com/spinnaker/orca/pull/4481
 https://github.com/spinnaker/rosco/pull/998
 
 For more information please see [the README](https://github.com/spinnaker/kork/blob/18d1c6e88597a9147851b37412ea38b3fd7032d5/kork-artifacts/src/main/java/com/netflix/spinnaker/kork/artifacts/README.md).
+
+### Rosco
+https://github.com/spinnaker/rosco/pull/986 adds in support for [Helmfile](https://helmfile.readthedocs.io/) as a bake manifest templating engine to Rosco. 
+
+For configuration please refer to the [deploy-helm user guide](https://github.com/spinnaker/spinnaker.io/blob/master/content/en/docs/guides/user/kubernetes-v2/deploy-helm/index.md#Helmfile)
```
Comment (thameezb) [inline content/en/docs/releases/next-release-preview/index.md:78]: This link will only be live once we merge to master. I am not sure the approach taken by the code owners to self-referential references

File: content/en/docs/guides/user/kubernetes-v2/deploy-helm/index.md [hunk 2]
```diff
@@ -123,6 +123,22 @@ of the same stage from above:
 }
 ```
 
+### Helmfile
+
+In addition to standard helm deployments, Spinnaker supports [Helmfile](https://helmfile.readthedocs.io/en/latest/) as a templating engine.
```
[the remark this answers is not in the document]
Comment (dbyron-sf) [reply content/en/docs/guides/user/kubernetes-v2/deploy-helm/index.md]: Yeah, there probably isn't a 100% bulletproof way,  https://github.com/spinnaker/rosco/blob/master/Dockerfile.slim#L7 will likely stop pointing to the helmfile version at some point, but so be it.

**Review by dbyron-sf — APPROVED** [review]

**Pull request merged**